code/modules/surgery/organs/lungs.dm

```diff
@@ -270,12 +270,60 @@
 			H.reagents.add_reagent("no2",1)
 
 		breath_gases[/datum/gas/nitryl][MOLES]-=gas_breathed
+
 	// Stimulum
 		gas_breathed = breath_gases[/datum/gas/stimulum][MOLES]
 		if (gas_breathed > gas_stimulation_min)
 			var/existing = H.reagents.get_reagent_amount("stimulum")
 			H.reagents.add_reagent("stimulum",max(0, 1 - existing))
 		breath_gases[/datum/gas/stimulum][MOLES]-=gas_breathed
+
+	// Miasma
+		if (breath_gases[/datum/gas/miasma])
+			var/miasma_pp = breath.get_breath_partial_pressure(breath_gases[/datum/gas/miasma][MOLES])
+
+			//Miasma sickness
+			if(prob(0.5 * miasma_pp))
+				var/datum/disease/advance/miasma_disease = new/datum/disease/advance
+				miasma_disease.symptoms = miasma_disease.GenerateSymptoms(1,3)
+				miasma_disease.try_infect(owner)
+
+			//Clearing out mood events if insufficient partial pressure
```

Comment:
remove these 2 checks. they should be handled in the switch statement

Reply:
you mind if I just keep the 2 moodlets and let another switch handle it so there's no conflicting 2 moodlets of the same type?

Reply:
whats your discord handle ill explain how moodlets work